plugin/src/main/groovy/com/novoda/staticanalysis/internal/CodeQualityConfigurator.groovy

```diff
@@ -78,17 +111,29 @@ abstract class CodeQualityConfigurator<T extends SourceTask, E extends CodeQuali
         }
     }
 
-    protected abstract void configureAndroidVariant(variant)
-
-    protected void configureJavaProject() {
-        project.tasks.withType(taskClass) { task ->
-            sourceFilter.applyTo(task)
-            task.exclude '**/*.kt'
+    protected static def javaCompile(variant) {
+        if (variant.hasProperty('javaCompileProvider')) {
+            variant.javaCompileProvider.get()
+        } else {
+            variant.javaCompile
         }
     }
 
+    protected final String getToolTaskNameFor(sourceSet) {
+        "$toolName${sourceSet.name.capitalize()}"
+    }
+
     protected abstract Class<T> getTaskClass()
 
-    protected abstract void configureReportEvaluation(T task, Violations violations)
+    protected abstract void createToolTaskForAndroid(sourceSet)
+
+    protected void configureToolTask(T task) {
+        sourceFilter.applyTo(task)
```

Comment:
Collected all common functionality here. It is automatically applied to all tool tasks thanks to `configureEach` in line 53 above.